resources/params.mjs

```diff
@@ -15,19 +18,42 @@ class Params {
     }
 
     _copyFromParams(searchParams) {
-        const viewportParam = searchParams.get("viewport");
-        if (viewportParam) {
+        if (searchParams.has("viewport")) {
+            const viewportParam = searchParams.get("viewport");
             const [width, height] = viewportParam.split("x");
             this.viewport.width = parseInt(width) || this.viewport.width;
             this.viewport.height = parseInt(height) || this.viewport.height;
-            if (this.viewport.width < 800 || this.viewport.height < 600) {
-                throw Error(`Invalid viewport param: ${viewportParam}`);
-            }
+            if (this.viewport.width < 800 || this.viewport.height < 600)
+                throw new Error(`Invalid viewport param: ${viewportParam}`);
+            searchParams.delete("viewport");
         }
         this.startAutomatically = searchParams.has("startAutomatically");
-        this.iterationCount = parseInt(searchParams.get("iterationCount")) || this.iterationCount;
-        if (this.iterationCount <= 1)
-            throw Error(`Invalid iterationCount param: ${this.iterationCount}`);
+        searchParams.delete("startAutomatically");
+        if (searchParams.has("iterationCount")) {
+            this.iterationCount = parseInt(searchParams.get("iterationCount")) || this.iterationCount;
+            if (this.iterationCount <= 1)
+                throw new Error(`Invalid iterationCount param: ${this.iterationCount}`);
+            searchParams.delete("iterationCount");
+        }
+
+        if (searchParams.has("unit")) {
+            this.unit = searchParams.get("unit");
+            if (!UNITS.includes(this.unit))
+                throw new Error(`Invalid unit=${this.unit}. Valid values are ${UNITS}`);
+            searchParams.delete("unit");
+        }
+
+        if (searchParams.has("suite") || searchParams.has("suites")) {
+            const value = searchParams.get("suites") || searchParams.get("suites");
```

Comment:
Isn't one of these supposed to be "suite", and not both "suites"?

Reply:
- I kept "suite" for backwards compatiblity (mayabe that's adding more confusion?)
- I'll add a warning if both "suite" and suites" are specified

Reply:
What I mean is that you have "suites" or "suites". Both of them are suites instead of one being "suite".

Reply:
Ah sorry, misread. DONE.